packages/maskbook/src/social-network-provider/facebook.com/UI/collectPosts.tsx

```diff
@@ -86,14 +88,21 @@ export function collectPostsFacebook(this: SocialNetworkUI) {
                 for (const url of images) {
                     info.postMetadataImages.add(url)
                     nextTypedMessage.push(makeTypedMessageImage(url))
+                    const post = await Services.Steganography.decodeImageUrl(url, {
+                        pass: info.postBy.value.toText(),
+                    })
+
+                    if (post.indexOf('🎼') === 0 || /https:\/\/.+\..+\/(\?PostData_v\d=)?%20(.+)%40/.test(post)) {
+                        info.iv.value = deconstructPayload(post, payloadDecoder).unwrap().iv
+                    }
                 }
                 // parse post content
                 info.postMessage.value = makeTypedMessageCompound(nextTypedMessage)
             }
             collectPostInfo()
```

Comment:
await?

Reply:
` new MutationObserverWatcher(posts).useForeach(async (node, key, metadata) => { //...` is not allowed, so here can't use `await collectPostInfo()`, we can do more investigation later. Currently fix the bug first.

Reply:
https://git.io/JkFKT How about make `foreach()` async

Reply:
No need to await, add `.catch(console.error)` is enough for me

Reply:
done